# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/Http2Session.cpp
# Commit: 00bb17e2a6a7
# Full Hash: 00bb17e2a6a73f9c66e861adae426818ff4d8caa
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2023-09-21 21:20:56
# Regressor Bug: 1784261
# File Overlap Count: 1
# Description:
#   Bug 1784261 - Add telemetry to understand the reason of connection closure, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188132
# ==============================================================================

diff -r e98350675c5a -r 00bb17e2a6a7 netwerk/protocol/http/Http2Session.cpp
--- a/netwerk/protocol/http/Http2Session.cpp	Thu Sep 21 08:42:26 2023 +0000
+++ b/netwerk/protocol/http/Http2Session.cpp	Thu Sep 21 08:59:03 2023 +0000
@@ -368,6 +368,7 @@
          this, pingTimeout));
     if ((now - mPingSentEpoch) >= pingTimeout) {
       LOG3(("Http2Session::ReadTimeoutTick %p Ping Timer Exhaustion\n", this));
+      mConnection->SetCloseReason(ConnectionCloseReason::IDLE_TIMEOUT);
       mPingSentEpoch = 0;
       if (isTrr) {
         // These must be set this way to ensure we gracefully restart all
@@ -2175,6 +2176,7 @@
     return self->SessionError(PROTOCOL_ERROR);
   }
 
+  self->mConnection->SetCloseReason(ConnectionCloseReason::GO_AWAY);
   self->mShouldGoAway = true;
   self->mGoAwayID = NetworkEndian::readUint32(self->mInputFrameBuffer.get() +
                                               kFrameHeaderBytes);